include/zephyr/bluetooth/classic/hfp_hf.h

```diff
@@ -53,108 +40,382 @@ struct bt_hfp_hf_cb {
 	 *  connection completes.
 	 *
 	 *  @param conn Connection object.
+	 *  @param hf HFP HF object.
 	 */
-	void (*connected)(struct bt_conn *conn);
+	void (*connected)(struct bt_conn *conn, struct bt_hfp_hf *hf);
 	/** HF disconnected callback to application
 	 *
 	 *  If this callback is provided it will be called whenever the
 	 *  connection gets disconnected, including when a connection gets
 	 *  rejected or cancelled or any error in SLC establishment.
+	 *  And the HFP HF object will be freed after the registered
+	 *  callback `disconnected` returned.
 	 *
-	 *  @param conn Connection object.
+	 *  @param hf HFP HF object.
 	 */
-	void (*disconnected)(struct bt_conn *conn);
+	void (*disconnected)(struct bt_hfp_hf *hf);
```

Comment:
Need consider adding a 'disconnected reason' to indicate the specific cause of the SLC disconnection?, similar to how the ACL BLE connection and SCO connection have a 'reason' parameter to indicate the reason for their disconnection."

Reply:
For ACL disconnection event, the reason is defined by Bluetooth SIG. For SLC disconnection, there is not such `reason`.